compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -294,6 +294,11 @@ genLoadLongConstant(TR::CodeGenerator * cg, TR::Node * node, int64_t value, TR::
       TR::Instruction * temp = cursor;
       cursor = generateRegLitRefInstruction(cg, TR::InstOpCode::LG, node, targetRegister, value, TR_GlobalValue, cond, cursor, base);
       }
+   else if (comp->compileRelocatableCode() && sym && (sym->isRecompilationCounter()))
+      {
+      TR::Instruction * temp = cursor;
+      cursor = generateRegLitRefInstruction(cg, TR::InstOpCode::LG, node, targetRegister, value, TR_BodyInfoAddress, cond, cursor, base);
+      }
```

Comment:
How were recompilation counters handled before? Perhaps we didn't generate them?

Reply:
We used to have different trees previously for recompilation counters which roughly looks like following 
```
iloadi
    loadaddr <recompilation_counter>
```
Relocatable code for `loadaddr` on Z is handled here : https://github.com/eclipse/omr/blob/22db7f45eb2d2593f711e87c59c7688064335443/compiler/z/codegen/OMRTreeEvaluator.cpp#L11528

Because of this change https://github.com/eclipse/openj9/pull/4311/files#diff-eeab0f6a82448f563e51b3e7529826f7 where we now have `iload <recompilation_counter>` and come to this function : https://github.com/eclipse/omr/blob/22db7f45eb2d2593f711e87c59c7688064335443/compiler/z/codegen/OMRTreeEvaluator.cpp#L279 which does not check if sym ref is of recompilation counter.